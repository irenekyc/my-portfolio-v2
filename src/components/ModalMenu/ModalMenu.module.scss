@-webkit-keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}
@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.ikModalMenu {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  text-align: center;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  display: none;
  width: 350px;

  &__about {
    -webkit-box-flex: 1;
    -ms-flex: 1;
    flex: 1;
    width: 100%;
    padding: 3rem 0;
    font-size: 1rem;
    -webkit-transform: translate3d(-100%, 0, 0);
    transform: translate3d(-100%, 0, 0);
    opacity: 0;
    min-width: 250px;
    display: flex;
    flex-direction: column;
    align-items: center;
    -webkit-transition: all 1s;
    transition: all 1s;
    min-width: 250px;
    padding: 3rem;
    -webkit-box-flex: 2;
    -ms-flex: 2;
    flex: 2;

    &__image {
      width: 100px;
      height: 100px;
      img {
        width: 100%;
        border-radius: 50%;
      }
    }
  }

  &__contact {
    min-width: 250px;
    padding: 3rem;
    -webkit-box-flex: 2;
    -ms-flex: 2;
    flex: 2;

    ul {
      list-style: none;
      padding: 0;
    }
    ul li {
      margin: 0.8rem 0;
      font-size: 1rem;
      -webkit-transform: translate3d(4rem, 0, 0);
      transform: translate3d(4rem, 0, 0);
      -webkit-transition: all 1s;
      transition: all 1s;
      opacity: 0;
    }
  }
  &__link {
    color: #eee;
    text-decoration: none;
  }
}

.ikModalMenu__active {
  .ikModalMenu {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
  }

  &.ikModalMenu__container {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    min-height: 100vh;
    background: rgba(58, 71, 80, 0.85);
    overflow-y: scroll;
  }

  .ikModalMenu__about {
    -webkit-animation: fadeIn 1s ease-in-out forwards;
    animation: fadeIn 1s ease-in-out forwards;
  }

  .ikModalMenu__contact ul li {
    -webkit-animation: fadeIn 0.5s ease-in-out forwards;
    animation: fadeIn 0.5s ease-in-out forwards;
    &:nth-child(1) {
      -webkit-animation-delay: 0.2s;
      animation-delay: 0.2s;
    }
    &:nth-child(2) {
      -webkit-animation-delay: 0.4s;
      animation-delay: 0.4s;
    }
    &:nth-child(3) {
      -webkit-animation-delay: 0.6s;
      animation-delay: 0.6s;
    }
    &:nth-child(4) {
      -webkit-animation-delay: 0.8s;
      animation-delay: 0.8s;
    }
    &:nth-child(5) {
      -webkit-animation-delay: 1s;
      animation-delay: 1s;
    }
  }
}
