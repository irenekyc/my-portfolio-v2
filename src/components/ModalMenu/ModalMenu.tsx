import React from "react";

const ModalMenu = () => {
  return (
    <div className="modal-menu">
      <div className="modal-menu-about">
        <div className="modal-menu-about-image">
          <img
            src="images/profile-thumbnail-2.jpg"
            alt="Irene K web developer"
          />
        </div>
        <h1> Irene K.</h1>
        <p> Web Developer</p>
      </div>
      <div className="modal-menu-contact">
        <ul>
          <li>
            <a
              href="https://irenekyc.github.io/irene-k/"
              rel="noopener noreferrer"
              target="_blank"
            >
              My Website
            </a>
          </li>
          <li>
            <a
              href="https://github.com/irenekyc"
              rel="noopener noreferrer"
              target="_blank"
            >
              {" "}
              Github
            </a>
          </li>
          <li>
            <a
              href="https://google.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              {" "}
              Resume
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/in/irene-chow-352662a8/"
              rel="noopener noreferrer"
              target="_blank"
            >
              {" "}
              LinkedIn{" "}
            </a>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default ModalMenu;
