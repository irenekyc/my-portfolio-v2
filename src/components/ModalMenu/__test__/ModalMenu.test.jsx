import React from "react";
import ModalMenu from "../index";
import { render, screen } from "@testing-library/react";

it("Render ModalMenu Section - should not show by default", () => {
  render(<ModalMenu showMenu={false} />);

  const modalMenu = screen.getByTestId("modal-menu");
  expect(modalMenu).not.toHaveClass("ikModalMenu__active");
});

it("Render ModalMenu Section - show", () => {
  render(<ModalMenu showMenu={true} />);

  const modalMenu = screen.getByTestId("modal-menu");
  expect(modalMenu).toHaveClass("ikModalMenu__active");
  const contactDetails = screen.getAllByTestId("modal-menu-contact-list");
  expect(contactDetails).toHaveLength(4);
  expect(modalMenu).toMatchSnapshot();
});
