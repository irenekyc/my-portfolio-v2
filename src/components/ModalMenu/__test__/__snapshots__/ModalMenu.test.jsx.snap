// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ModalMenu Section - show 1`] = `
<section
  class="ikModalMenu__container ikModalMenu__active"
  data-testid="modal-menu"
>
  <div
    class="ikModalMenu"
  >
    <div
      class="ikModalMenu__about"
      data-testid="modal-menu-about"
    >
      <div
        class="ikModalMenu__about__image"
      >
        <img
          alt="Irene K web developer"
          src="images/profile-thumbnail-2.jpg"
        />
      </div>
      <h2>
         Irene K.
      </h2>
      <p>
         Web Developer
      </p>
    </div>
    <div
      class="ikModalMenu__contact"
    >
      <ul>
        <li
          data-testid="modal-menu-contact-list"
        >
          <a
            class="ikModalMenu__link"
            href="https://irenekyc.github.io/irene-k/"
            rel="noopener noreferrer"
            target="_blank"
          >
            My Website
          </a>
        </li>
        <li
          data-testid="modal-menu-contact-list"
        >
          <a
            class="ikModalMenu__link"
            href="https://github.com/irenekyc"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            Github
          </a>
        </li>
        <li
          data-testid="modal-menu-contact-list"
        >
          <a
            class="ikModalMenu__link"
            href="https://drive.google.com/file/d/10QU9exsCAFiL1P2sM-ljihuGE-cfCeIV/view?usp=sharing"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            Resume
          </a>
        </li>
        <li
          data-testid="modal-menu-contact-list"
        >
          <a
            class="ikModalMenu__link"
            href="https://www.linkedin.com/in/irene-chow-352662a8/"
            rel="noopener noreferrer"
            target="_blank"
          >
             
            LinkedIn
             
          </a>
        </li>
      </ul>
    </div>
  </div>
</section>
`;
